File: src/main/java/demo/bfims/Entities/Inventory/Accessory/AccessoryItem.java
package demo.bfims.Entities.Inventory.Accessory;

import demo.bfims.Entities.Inventory.Publication.Item;
import demo.bfims.Enums.AccessoryItemStatus;
import demo.bfims.Enums.ItemType;
import demo.bfims.Interfaces.Purchaseable;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class AccessoryItem extends Item implements Purchaseable {
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "accessory_id")
    private Accessory accessory;
    private AccessoryItemStatus accessoryItemStatus;

    public AccessoryItem() {
        this.setItemType(ItemType.ACCESSORY_ITEM);
    }

    public AccessoryItemStatus getAccessoryItemStatus() {
        return accessoryItemStatus;
    }

    public void setAccessoryItemStatus(AccessoryItemStatus accessoryItemStatus) {
        this.accessoryItemStatus = accessoryItemStatus;
    }

    public Accessory getAccessory() {
        return accessory;
    }

    public void setAccessory(Accessory accessory) {
        this.accessory = accessory;
    }

    @Override
    public Double getPurchasePrice() {
        return this.getAccessory().getPrice();
    }

    @Override
    public void setPurchasePrice(Double purchasePrice) {
        this.getAccessory().setPrice(purchasePrice);
    }

    @Override
    public String toString() {
        return "AccessoryItem{" +
                "accessory=" + accessory +
                ", accessoryItemStatus=" + accessoryItemStatus +
                "} " + super.toString();
    }
}

File: frontend/src/app/DTOs/Accessory/AccessoryItem.ts
import {Item} from '../Inventory/Item';
import {Accessory} from './Accessory';
import {AccessoryItemStatus} from '../../Enums/AccessoryItemStatus';
import {ItemType} from '../../Enums/ItemType';

export class AccessoryItem extends Item {
  public accessory: Accessory;
  public accessoryItemStatus: AccessoryItemStatus;

  constructor(...data: any) {
    data = data[0];
    super(data);
    if (data) {
      this.accessory = data["accessory"] || null;
      this.accessoryItemStatus = data["accessoryItemStatus"] || null;
    }
      this.itemType = ItemType.ACCESSORY_ITEM;
  }
}
